room/runtime/src/main/java/androidx/room/SQLiteCopyOpenHelper.java

```diff
@@ -211,10 +207,66 @@ private void copyDatabaseFile(File destinationFile) throws IOException {
                     + destinationFile.getAbsolutePath());
         }
 
+        // Temporarily open intermediate file database using FrameworkSQLiteOpenHelper and call
+        // open pre-packaged callback. If it fails then intermediate file won't be copied making
+        // invoking pre-packaged callback a transactional operation.
+        callOnOpenPrepackagedDatabaseCallbacks(intermediateFile, writable);
+
         if (!intermediateFile.renameTo(destinationFile)) {
             throw new IOException("Failed to move intermediate file ("
                     + intermediateFile.getAbsolutePath() + ") to destination ("
                     + destinationFile.getAbsolutePath() + ").");
         }
     }
+
+    private Integer readDatabaseVersion(File databaseFile) {
```

Comment:
Not sure this method is necessary, in `callOnOpenPrepackagedDatabaseCallbacks` where it is now newly used we should use the `DBUtil` version so that we can add the `IOException` as the cause to the `RuntimeException`.